### PR TITLE
Update HESA ethnicities lookup for 2021-2022

### PR DESCRIPTION
**Pull request by steventux**
## Context

We lookup HESA ethnicity codes based on the current recruitment year as some values are particular to specific years.
The data remains unchanged for the 2021-2022 cycle but we don't currently support this year.
See https://www.hesa.ac.uk/collection/c21053/revision_history for changes.

## Changes proposed in this pull request

Refactored the lookup method to allow a dynamic lookup based on the year.

## Link to Trello card

https://trello.com/c/6rTx4hAJ/4185-investigation-review-hesa-data-cycle-awareness-and-storage

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: config/initializers/hesa_ethnicities.rb [hunk 1]
```diff
@@ -19,29 +19,34 @@ module HesaEthnicityValues
...
+    ['80', HesaEthnicityValues::OTHER_ETHNIC],
+    ['90', HesaEthnicityValues::NOT_KNOWN],
+    ['98', HesaEthnicityValues::INFORMATION_REFUSED],
+  ].freeze
 
-# Two codes have been dropped in 2020/21
-# https://www.hesa.ac.uk/collection/c20053/e/ethnic
-HESA_ETHNICITIES_2020_2021 = HESA_ETHNICITIES_2019_2020
-                               .reject { |ethnicity| %w[90 98].include?(ethnicity.first) }
+  # Two codes have been dropped in 2020/21
```
Comment (benswannack) [inline config/initializers/hesa_ethnicities.rb:45]: Just double checking this claim, it looks like `90` and `98` are still present in all of the docs (2020 though to 2022)

Comment (steventux) [reply config/initializers/hesa_ethnicities.rb:45]: @TobyRet can you shed any light on this? The HESA data sets seem to contain these two codes for all revisions I can find.

Comment (steventux) [reply config/initializers/hesa_ethnicities.rb]: @benswannack having chatted with Toby we're agreed the documentation no longer omits these values, not sure why but I've echoed this in https://github.com/DFE-Digital/apply-for-teacher-training/pull/5524/commits/9a00debec76d6c5384e290adcd313fb6c6e21325

**Review by benswannack — APPROVED** [review]
Looks good :)

**Pull request merged**